### PR TITLE
SQL-707: Update getTypeInfo() result set to be ordered according to the specification

### PR DESCRIPTION
**Pull request by EthanHardyMongo**
I changed the ordering of the result set to be ordered by DATA_TYPE (from least to greatest) if the DATA_TYPE is not equal to 1111. Additionally, I sorted all the Types with DATA_TYPE == 1111 alphabetically and put them after the types with DATA_TYPE != 1111.

### CODE REVIEW

**Review by mattChiaravalloti — APPROVED** [review]
✨ Excellent comment ✨  LGTM

File: resources/integration_test/tests/dbmd_constant_result_sets.yml [hunk 1]
```diff
@@ -32,94 +32,94 @@ tests:
     db: integration_test
     meta_function: [ getTypeInfo ]
     expected_sql_type: [ LONGVARCHAR, INTEGER, INTEGER, LONGVARCHAR, LONGVARCHAR, LONGVARCHAR,
-                         INTEGER, BOOLEAN, INTEGER, BOOLEAN, BOOLEAN, BOOLEAN, LONGVARCHAR, INTEGER, INTEGER, INTEGER,
-                         INTEGER, INTEGER ]
+                          INTEGER, BOOLEAN, INTEGER, BOOLEAN, BOOLEAN, BOOLEAN, LONGVARCHAR, INTEGER,
```
Comment (bucaojit) [inline resources/integration_test/tests/dbmd_constant_result_sets.yml:35]: Just curious, did you use a yaml formatting tool to correct this format?

**Review by bucaojit — APPROVED** [review]
LGTM

Ordering looks correct, thanks for adding the comment also to explain the ordering according to the javadocs

File: resources/integration_test/tests/dbmd_constant_result_sets.yml (hunk 1 above)
Comment (bucaojit) [reply resources/integration_test/tests/dbmd_constant_result_sets.yml:35]: Ok thanks for letting me know.  No, we don't have another tool for the yaml so I was wondering how the formatting changed.  👍

**Comment by EthanHardyMongo** [comment]
evergreen merge

**Pull request merged**